simpleclient_dropwizard/src/main/java/io/prometheus/client/dropwizard/DropwizardExports.java

```diff
@@ -9,80 +11,88 @@
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import java.util.regex.Pattern;
 
 /**
  * Collect Dropwizard metrics from a MetricRegistry.
  */
 public class DropwizardExports extends io.prometheus.client.Collector implements io.prometheus.client.Collector.Describable {
-    private MetricRegistry registry;
     private static final Logger LOGGER = Logger.getLogger(DropwizardExports.class.getName());
+    private MetricRegistry registry;
+    private SampleBuilder sampleBuilder;
 
     /**
+     * Creates a new DropwizardExports with a {@link DefaultSampleBuilder}.
+     *
      * @param registry a metric registry to export in prometheus.
      */
     public DropwizardExports(MetricRegistry registry) {
         this.registry = registry;
+        this.sampleBuilder = new DefaultSampleBuilder();
     }
 
     /**
-     * Export counter as Prometheus <a href="https://prometheus.io/docs/concepts/metric_types/#gauge">Gauge</a>.
+     * @param registry      a metric registry to export in prometheus.
+     * @param sampleBuilder sampleBuilder to use to create prometheus samples.
      */
-    List<MetricFamilySamples> fromCounter(String dropwizardName, Counter counter) {
-        String name = sanitizeMetricName(dropwizardName);
-        MetricFamilySamples.Sample sample = new MetricFamilySamples.Sample(name, new ArrayList<String>(), new ArrayList<String>(),
-                new Long(counter.getCount()).doubleValue());
-        return Arrays.asList(new MetricFamilySamples(name, Type.GAUGE, getHelpMessage(dropwizardName, counter), Arrays.asList(sample)));
+    public DropwizardExports(MetricRegistry registry, SampleBuilder sampleBuilder) {
+        this.registry = registry;
+        this.sampleBuilder = sampleBuilder;
     }
 
-    private static String getHelpMessage(String metricName, Metric metric){
+    private static String getHelpMessage(String metricName, Metric metric) {
         return String.format("Generated from Dropwizard metric import (metric=%s, type=%s)",
                 metricName, metric.getClass().getName());
     }
 
+    /**
+     * Export counter as Prometheus <a href="https://prometheus.io/docs/concepts/metric_types/#gauge">Gauge</a>.
+     */
+    List<MetricFamilySamples> fromCounter(String dropwizardName, Counter counter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(dropwizardName, "", new ArrayList<String>(), new ArrayList<String>(),
+                new Long(counter.getCount()).doubleValue());
+        return Arrays.asList(new MetricFamilySamples(sample.name, Type.GAUGE, getHelpMessage(dropwizardName, counter), Arrays.asList(sample)));
```

Comment:
You need to return all of a given metric in the same metric family

Reply:
Could you please explain what is wrong? I am setting the metric name with the same value as the sample name, like it was before.

Reply:
Now you can have multiple metrics with the same name, so they need to be grouped together under the same metric family.

Reply:
I see what you mean, thank you.
It might also happen that different types of metrics end up under the same metric name. I reckon we should throw an `IllegalStateException` in this situation. What do you think?

Reply:
That's not something I've seen in practice, I wouldn't worry about it.

Reply:
OK, I've added some grouping logic in the collector.